.github/workflows/pypi_publish.yml

```diff
@@ -0,0 +1,170 @@
+name: Publish to PyPi
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: "Version to publish (leave empty for release version)"
+        required: false
+        type: string
+      dry_run:
+        description: "Dry run - publish to Test-PyPi repo only"
+        required: false
+        default: true
+        type: boolean
+
+permissions:
+  contents: read
+
+jobs:
+  build-wheels:
+    name: Build platform specific wheels
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
```

Comment:
What about the arm targets?

Reply:
there was a compilation error, that doesn't seem trivial, which is why I removed them for now. Needs more time to fix this. I can create an issue and address it specifically.